WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt

```diff
@@ -0,0 +1,363 @@
+            }
+
+            itemView.addView(customView)
+        }
+
+        val position = AppPrefs.getMainPageIndex()
+        currentPosition = position
+    }
+
+    /*
+     * uses reflection to disable "shift mode" so the item are equal width
+     */
+    @SuppressLint("RestrictedApi")
+    private fun disableShiftMode() {
+        val menuView = getChildAt(0) as BottomNavigationMenuView
+        try {
+            val shiftingMode = menuView.javaClass.getDeclaredField("mShiftingMode")
+            shiftingMode.isAccessible = true
+            shiftingMode.setBoolean(menuView, false)
+            shiftingMode.isAccessible = false
+            for (i in 0 until menuView.childCount) {
+                val item = menuView.getChildAt(i) as BottomNavigationItemView
+                item.setShiftingMode(false)
+                // force the view to update
+                item.setChecked(item.itemData.isChecked)
+            }
+        } catch (e: NoSuchFieldException) {
+            AppLog.e(T.MAIN, "Unable to disable shift mode", e)
+        } catch (e: IllegalAccessException) {
+            AppLog.e(T.MAIN, "Unable to disable shift mode", e)
+        }
+    }
+
+    private fun assignNavigationListeners(assign: Boolean) {
+        setOnNavigationItemSelectedListener(if (assign) this else null)
+        setOnNavigationItemReselectedListener(if (assign) this else null)
+    }
+
+    override fun onNavigationItemSelected(item: MenuItem): Boolean {
+        val position = getPositionForItemId(item.itemId)
+        return if (position == PAGE_NEW_POST) {
+            handlePostButtonClicked()
+            false
+        } else {
+            setCurrentPosition(position, false)
+            mListener!!.onPageChanged(position)
+            true
+        }
+    }
+
+    private fun handlePostButtonClicked() {
+        val postView = getItemView(PAGE_NEW_POST)
+
+        // animate the button icon before telling the listener the post button was clicked - this way
+        // the user sees the animation before the editor appears
+        AniUtils.startAnimation(postView, R.anim.navbar_button_scale, object : Animation.AnimationListener {
+            override fun onAnimationStart(animation: Animation) {
+                // noop
+            }
+
+            override fun onAnimationEnd(animation: Animation) {
+                mListener!!.onNewPostButtonClicked()
+            }
+
+            override fun onAnimationRepeat(animation: Animation) {
+                // noop
+            }
+        })
+    }
+
+    override fun onNavigationItemReselected(item: MenuItem) {
+        // scroll the active fragment's contents to the top when user re-taps the current item
+        val position = getPositionForItemId(item.itemId)
+        if (position != PAGE_NEW_POST) {
+            val fragment = mNavAdapter!!.getFragment(position)
+            if (fragment is OnScrollToTopListener) {
+                (fragment as OnScrollToTopListener).onScrollToTop()
```

Comment:
you could use `(mNavAdapter?.getFragment(position) as? OnScrollToTopListener)?.onScrollToTop()` instead of these 3 lines. I generally find safe cast `as?` as a handy tool